plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/gitlabMergeRequest.ts

```diff
@@ -0,0 +1,160 @@
+import { createTemplateAction } from '@backstage/plugin-scaffolder-backend';
+import { readFile } from 'fs-extra';
+import { Gitlab } from '@gitbeaker/node';
+import path from 'path';
+import globby from 'globby';
+import { CommitAction } from '@gitbeaker/core/dist/types/services/Commits';
+import {  CreateMergeRequestOptions } from '@gitbeaker/core/dist/types/services/MergeRequests';
+import { ScmIntegrationRegistry } from '@backstage/integration';
+import { InputError } from '@backstage/errors';
+import { parseRepoUrl } from './util';
+
+export type GitlabMergeRequestActionInput = {
+	projectid: string;
+	repoUrl: string;
+	title: string;
+	description: string;
+	destinationBranch: string;
+	targetPath: string;
+  };
+
+export const createPublishGitlabMergeRequestAction = (options: {
+	integrations: ScmIntegrationRegistry;
+  }) => {
+	const { integrations } = options;
+
+	return createTemplateAction<GitlabMergeRequestActionInput>({
+	  id: 'publish:gitlab-merge-request',
+	  schema: {
+		input: {
+		  required: ['projectid', 'repoUrl', 'targetPath'],
+		  type: 'object',
+		  properties: {
+			repoUrl: {
+				type: 'string',
+				title: 'Repository Location',
+				description: `Accepts the format 'gitlab.com/group_name/project_name' where 'project_name' is the repository name and 'group_name' is a group or username`,
+			},
+			projectid: {
```

Comment:
Maybe we could leverage the same functionality added in https://github.com/loblaw-sre/backstage-plugin-gitlab/pull/23 and dont require Project ID, making it easier on the end user?

Reply:
@Balasundaram I agree with @dublx, that will keep the consistency that project ID is not a mandatory field.